deepspeed/inference/engine.py

```diff
@@ -305,29 +304,6 @@ def _init_quantization_setting(self, quantization_setting):
             f"mlp_extra_grouping = {self.mlp_extra_grouping}, "
             f"quantize_groups = {self.quantize_groups}", [0])
 
-    # TODO: remove this function and add this functionality to pydantic config checking
```

Comment:
@nelyahu - we never added this functionality/checking to the pydantic config, let me take a stab at that before we merge this PR?